calendar/calendar-impl/impl/src/java/org/sakaiproject/calendar/impl/BaseExternalCalendarSubscriptionService.java

```diff
@@ -314,10 +318,36 @@ public Calendar getCalendarSubscription(String reference)
 		m_log.debug("ExternalCalendarSubscriptionService.getCalendarSubscription("
 				+ reference + ")");
 		m_log.debug(" |-> subscriptionUrl: " + subscriptionUrl);
+		
+		String currentSessionUserId = m_sessionManager.getCurrentSessionUserId();
+		
+		//External references will not work for anonymous users, no idea how to track circular references for them.
+		if ((currentSessionUserId==null)||(currentSessionUserId.isEmpty())) return null;
```

Comment:
I have found in StackOverflow that "".equals(currentSessionUserId) covers both null and empty string cases in a single check. I change it.